src/app/api/user/[userId]/bookAppointment/route.js

```diff
@@ -0,0 +1,34 @@
+import { NextResponse } from "next/server"
+import Appointment from "../../../../../../models/Appointment";
+import User from "../../../../../../models/User";
+import dbConnect from "../../../../../../utils/dbConnect";
+
+export const POST = async(request, { params }) => {
+  try {
+    await dbConnect();
+
+    const body = await request.json();
+    const { userId } = await params;
+
+    const user = await User.findById(userId);
+
+    if (!user) return NextResponse.json({error: "A user does not exist with this id."}, {status: 404});

```

Comment:
I like short error messages. Have you thought about something simple like "Oops! User does not exist!"

At the same time though, this is a specific error message for the backend, and can help with testing.